src/main/java/org/jenkinsci/plugins/mesos/MesosOneOffSlave.java

```diff
@@ -0,0 +1,95 @@
+package org.jenkinsci.plugins.mesos;
+
+
+import hudson.Extension;
+import hudson.Launcher;
+import hudson.model.BuildListener;
+import hudson.model.AbstractBuild;
+import hudson.model.AbstractProject;
+import hudson.slaves.OfflineCause;
+import hudson.tasks.BuildWrapper;
+import hudson.tasks.BuildWrapperDescriptor;
+
+import java.io.IOException;
+import java.util.logging.Logger;
+
+import org.kohsuke.stapler.DataBoundConstructor;
+
+/**
+ This file is part of the JCloud Jenkins Plugin. (https://github.com/jenkinsci/jclouds-plugin)
+ commit id 20c6ca884abb172b27d0af82545c8915ce08f618.
+
+ This file was modified to work with the Mesos Jenkins plugin and based off the JClouds Jenkins Plugin.
+
+ According to the Jenkins Plugin guide (https://wiki.jenkins-ci.org/display/JENKINS/Before+starting+a+new+plugin),
+ If no license is defined in the form of a header, LICENSE file, or in the license section, the code is assumed to be
+ under The MIT License.
+
+ The MIT License (MIT)
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in
+ all copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+ */
+public class MesosOneOffSlave extends BuildWrapper {
+    private static final Logger LOGGER = Logger.getLogger(MesosOneOffSlave.class.getName());
+
+    @DataBoundConstructor
+    public MesosOneOffSlave() {
+    }
+
+    @Override
+    @SuppressWarnings("rawtypes")
+    public Environment setUp(AbstractBuild build, Launcher launcher, final BuildListener listener) {
+        if (MesosComputer.class.isInstance(build.getExecutor().getOwner())) {
+            final MesosComputer c = (MesosComputer) build.getExecutor().getOwner();
+            return new Environment() {
+                @Override
+                public boolean tearDown(AbstractBuild build, final BuildListener listener) throws IOException, InterruptedException {
+                    LOGGER.warning("Single-use slave " + c.getName() + " getting torn down.");
```

Comment:
Lets stick with one terminology. either "one off slave" or "single use slave" for file names, variable names and log messages.